controllers/volumereplication_controller.go

```diff
@@ -317,6 +328,50 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 		WithEventFilter(pred).Complete(r)
 }
 
+func (r *VolumeReplicationReconciler) waitForCrds() error {
+
+	logger := r.Log.WithName("checkingDependencies")
+
+	err := r.waitForVolumeReplicationResource(logger, VolumeReplicationClass)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplicationClass")
+		return err
+	}
+
+	err = r.waitForVolumeReplicationResource(logger, VolumeReplication)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplication")
+		return err
+	}
+	return nil
+}
+
+func (r *VolumeReplicationReconciler) waitForVolumeReplicationResource(logger logr.Logger, resourceName string) error {
+
+	isVolumeReplicationResourceExists := false
+	for ! isVolumeReplicationResourceExists {
+		unstructuredResource := &unstructured.UnstructuredList{}
```

Comment:
this need to be initialized once move this out of for loop.

Reply:
done